barman/infofile.py

```diff
@@ -897,6 +918,8 @@ def walk_backups_tree(self):
                     backup_id=child_backup_id,
                 )
                 yield from backup_info.walk_backups_tree()
+        if return_self is False and self.parent_backup_id is None:
```

Comment:
```suggestion
        if not return_self:
```

This should be enough I guess?

This way we also allow bringing all the children of an intermediate child in the backup tree without returning the own child itself.